src/plugins/es_ui_shared/static/forms/hook_form_lib/components/use_multi_fields.test.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useState } from 'react';
+
+import { registerTestBed } from '../shared_imports';
+import { FieldHook } from '../types';
+import { useForm } from '../hooks/use_form';
+import { Form } from './form';
+import { UseMultiFields } from './use_multi_fields';
+
+describe('<UseMultiFields />', () => {
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  const fields = {
+    foo: { path: 'foo' },
+    bar: { path: 'bar' },
+  };
+
+  const TestComp = ({ onFields }: { onFields: (fields: { [x: string]: FieldHook }) => void }) => {
+    const { form } = useForm();
+    const [stateFields, setStateFields] = useState<{ [key: string]: any }>(fields);
+
+    const changeStateFields = () => {
+      // We'll make sure that if other fields are passed down after the initial
+      // rendering of <UseMultiField /> the change does not create new FieldHook as that
+      // would breal the **order** of hooks declared inside <UseMultiFields />
+
+      setStateFields({
+        aaa: { path: 'aaa' }, // we add this field that will come first when sorting() A-Z
+        ...fields,
+      });
+    };
+
+    return (
+      <Form form={form}>
+        <UseMultiFields fields={stateFields}>
+          {(hookFields) => {
+            onFields(hookFields);
+            return null;
```

Comment:
Why is `return null` here necessary?

Reply:
A habit that I have. I like to return null when there is no content, it makes it explicit.